cosmos/providers/dbt/core/operators.py

```diff
@@ -3,13 +3,14 @@
 import os
 import signal
 from pathlib import Path
-from typing import List, Sequence, TypedDict
+from typing import List, Sequence
 
 import yaml
 from airflow.compat.functools import cached_property
 from airflow.exceptions import AirflowException, AirflowSkipException
 from airflow.hooks.subprocess import SubprocessHook
 from airflow.models.baseoperator import BaseOperator
+from airflow.typing_compat import TypedDict
```

Comment:
I'm not sure how quickly Airflow will deprecate the `TypedDict` import from this module after [Python 3.7. hits EOL in 4 months](https://endoflife.date/python). Basically Airflow [will drop Python support versions in the next minor or major Airflow version after a Python version reaches EOL](https://airflow.apache.org/docs/apache-airflow/stable/installation/supported-versions.html#support-for-python-and-kubernetes-versions). @jlaneve not sure how long you'd like to support Python 3.7, but it might make sense to port over the compat logic to a native module in Cosmos. WDYT?

Reply:
FWIW, Python 3.7 support has been removed from Airflow `main` and presumably starting with 2.7.0 only support Python 3.8+.